stumpy/stump.py

```diff
@@ -177,38 +199,46 @@ def _compute_diagonal(
                 #     - (T_B[i + k - 1] - M_T_m_1[i + k]) * (T_A[i - 1] - μ_Q_m_1[i])
                 # )
                 cov = cov + constant * (
-                    cov_a[i + k] * cov_b[i] - cov_c[i + k] * cov_d[i]
+                    cov_a[i + g] * cov_b[i] - cov_c[i + g] * cov_d[i]
                 )
 
-            if T_B_subseq_isfinite[i + k] and T_A_subseq_isfinite[i]:
+            if T_B_subseq_isfinite[i + g] and T_A_subseq_isfinite[i]:
                 # Neither subsequence contains NaNs
-                if T_B_subseq_isconstant[i + k] or T_A_subseq_isconstant[i]:
+                if T_B_subseq_isconstant[i + g] or T_A_subseq_isconstant[i]:
                     pearson = 0.5
                 else:
-                    pearson = cov * Σ_T_inverse[i + k] * σ_Q_inverse[i]
+                    pearson = cov * Σ_T_inverse[i + g] * σ_Q_inverse[i]
 
-                if T_B_subseq_isconstant[i + k] and T_A_subseq_isconstant[i]:
+                if T_B_subseq_isconstant[i + g] and T_A_subseq_isconstant[i]:
                     pearson = 1.0
 
                 if pearson > ρ[thread_idx, i, 0]:
-                    ρ[thread_idx, i, 0] = pearson
-                    I[thread_idx, i, 0] = i + k
+                    idx = np.searchsorted(ρ[thread_idx, i], pearson)
+                    ρ[thread_idx, i, : idx - 1] = ρ[thread_idx, i, 1:idx]
```

Comment:
Because the idx that is discovered is the insert position of  array `ρ[thread_idx, i]`. But, note that we are discarding the first element (reminder: this is ascendingly-sorted pearson correlation and we discard the first element when we want to update it). So, when  we discard an element, the insertion point becomes `idx - 1`. 

I will think about it to see if I can rewrite it.

Reply:
Got it. Please add a small comment with the reminder and I think that will be sufficient :)